### PR TITLE
Update kube dependency to 1.14 and CSI spec to v1.1.0

### PR DESCRIPTION
**Pull request by humblec**
#Fix https://github.com/ceph/ceph-csi/issues/368
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@Madhu-1 @gman0 @ShyamsundarR PTAL

File: pkg/rbd/rbd.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ func (r *Driver) Run(driverName, nodeID, endpoint, configRoot string, containeri
 		csi.ControllerServiceCapability_RPC_CREATE_DELETE_SNAPSHOT,
 		csi.ControllerServiceCapability_RPC_LIST_SNAPSHOTS,
 		csi.ControllerServiceCapability_RPC_CLONE_VOLUME,
+		csi.ControllerServiceCapability_RPC_EXPAND_VOLUME,
```
Comment (ShyamsundarR) [inline pkg/rbd/rbd.go]: @humblec I did not understand why we are adding this capability when we actually do not have it?

File: pkg/cephfs/controllerserver.go [hunk 2]
```diff
@@ -189,3 +189,7 @@ func (cs *ControllerServer) ValidateVolumeCapabilities(
 		},
 	}, nil
 }
+
+func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
```
Comment (ShyamsundarR) [inline pkg/cephfs/controllerserver.go]: The default unimplemented RPCs goto controllerserver-default.go and nodeserver-default.go etc. under `pkg/csi-common`, is there a reason to put these here?

File: pkg/cephfs/driver.go [hunk 3]
```diff
@@ -127,6 +127,10 @@ func (fs *Driver) Run(driverName, nodeID, endpoint, volumeMounter, mountCacheDir
 		csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER,
 	})
 
+	fs.cd.AddControllerServiceCapabilities([]csi.ControllerServiceCapability_RPC_Type{
```
Comment (Madhu-1) [inline pkg/cephfs/driver.go]: this is also not needed  as we are not supporting expand volume

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR it was my mistake. Corrected it.  PTAL

File: pkg/cephfs/driver.go (hunk 3 above)
Comment (humblec) [reply pkg/cephfs/driver.go]: Done

File: pkg/rbd/controllerserver.go [hunk 4]
```diff
@@ -626,3 +626,8 @@ func (cs *ControllerServer) ListSnapshots(ctx context.Context, req *csi.ListSnap
 		Entries: entries,
 	}, nil
 }
+
+func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
```
Comment (ShyamsundarR) [inline pkg/rbd/controllerserver.go]: afaict we do not need these, as the default files already have the required definitions.

File: pkg/rbd/nodeserver.go [hunk 5]
```diff
@@ -293,3 +293,8 @@ func parseFindMntResolveSource(out string) (string, error) {
 	}
 	return "", fmt.Errorf("parseFindMntResolveSource: %s doesn't match to any expected findMnt output", out)
 }
+
+func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
```
Comment (ShyamsundarR) [inline pkg/rbd/nodeserver.go]: afaict we do not need these, as the default files already have the required definitions.

File: Gopkg.toml [hunk 6]
```diff
@@ -11,15 +11,15 @@
   version = "1.10.0"
 
 [[constraint]]
-  version = "kubernetes-1.13.3"
+  version = "kubernetes-1.14.0"
```
Comment (ShyamsundarR) [inline Gopkg.toml]: I am surprised that these changes did not pull in additional vendor deps or update vendor dependencies. I am not familar with the tool, but usual updates to versions update a lot of vendor files. @Madhu-1 please check and see if this is an issue or just my misunderstanding.

**Comment by Madhu-1** [comment]
Ran `dep check` looks like `Gopkg.lock` and `Gopkg.toml` are  not in sync
```
[🎩]mrajanna@localhost ceph-csi $]dep check
# Gopkg.lock is out of sync:
github.com/gogo/protobuf/proto: in Gopkg.lock's input-imports, but neither imported nor required
github.com/gogo/protobuf/sortkeys: in Gopkg.lock's input-imports, but neither imported nor required
github.com/golang/protobuf/descriptor: in Gopkg.lock's input-imports, but neither imported nor required
github.com/golang/protobuf/proto: in Gopkg.lock's input-imports, but neither imported nor required
github.com/golang/protobuf/protoc-gen-go/descriptor: in Gopkg.lock's input-imports, but neither imported nor required
github.com/golang/protobuf/ptypes/any: in Gopkg.lock's input-imports, but neither imported nor required
github.com/golang/protobuf/ptypes/duration: in Gopkg.lock's input-imports, but neither imported nor required
github.com/golang/protobuf/ptypes/wrappers: in Gopkg.lock's input-imports, but neither imported nor required
github.com/google/btree: in Gopkg.lock's input-imports, but neither imported nor required
github.com/google/gofuzz: in Gopkg.lock's input-imports, but neither imported nor required
github.com/google/uuid: in Gopkg.lock's input-imports, but neither imported nor required
github.com/googleapis/gnostic/OpenAPIv2: in Gopkg.lock's input-imports, but neither imported nor required
github.com/googleapis/gnostic/compiler: in Gopkg.lock's input-imports, but neither imported nor required
github.com/googleapis/gnostic/extensions: in Gopkg.lock's input-imports, but neither imported nor required
github.com/gregjones/httpcache: in Gopkg.lock's input-imports, but neither imported nor required
github.com/gregjones/httpcache/diskcache: in Gopkg.lock's input-imports, but neither imported nor required
github.com/json-iterator/go: in Gopkg.lock's input-imports, but neither imported nor required
github.com/modern-go/concurrent: in Gopkg.lock's input-imports, but neither imported nor required
github.com/modern-go/reflect2: in Gopkg.lock's input-imports, but neither imported nor required
github.com/petar/GoLLRB/llrb: in Gopkg.lock's input-imports, but neither imported nor required
github.com/peterbourgon/diskv: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/crypto/ssh/terminal: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/net/context/ctxhttp: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/net/http2: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/net/http2/hpack: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/net/idna: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/oauth2: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/sys/unix: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/collate: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/internal/gen: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/internal/ucd: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/language: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/transform: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/unicode/bidi: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/text/unicode/rangetable: in Gopkg.lock's input-imports, but neither imported nor required
golang.org/x/time/rate: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/appengine/internal/base: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/appengine/internal/datastore: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/appengine/internal/log: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/appengine/internal/remote_api: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/appengine/urlfetch: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/genproto/googleapis/rpc/status: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/balancer: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/balancer/base: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/binarylog/grpc_binarylog_v1: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/connectivity: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/credentials: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/encoding: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/grpclog: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/internal/backoff: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/internal/channelz: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/internal/grpcrand: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/internal/syscall: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/keepalive: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/metadata: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/peer: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/resolver: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/stats: in Gopkg.lock's input-imports, but neither imported nor required
google.golang.org/grpc/tap: in Gopkg.lock's input-imports, but neither imported nor required
gopkg.in/inf.v0: in Gopkg.lock's input-imports, but neither imported nor required
gopkg.in/yaml.v2: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/admissionregistration/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/admissionregistration/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/apps/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/apps/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/apps/v1beta2: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/auditregistration/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/authentication/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/authentication/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/authorization/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/authorization/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/autoscaling/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/autoscaling/v2beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/autoscaling/v2beta2: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/batch/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/batch/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/batch/v2alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/certificates/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/coordination/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/events/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/extensions/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/networking/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/policy/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/rbac/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/rbac/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/rbac/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/scheduling/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/scheduling/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/settings/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/storage/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/storage/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/api/storage/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/api/meta: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/conversion: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/fields: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/schema: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/serializer: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/serializer/json: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/serializer/protobuf: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/serializer/recognizer: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/serializer/streaming: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/runtime/serializer/versioning: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/types: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/clock: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/errors: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/framer: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/json: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/net: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/runtime: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/validation/field: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/util/yaml: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/version: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/apimachinery/pkg/watch: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/kubernetes/scheme: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/pkg/apis/clientauthentication: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/pkg/apis/clientauthentication/v1alpha1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/pkg/apis/clientauthentication/v1beta1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/pkg/version: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/plugin/pkg/client/auth/exec: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/rest/watch: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/tools/clientcmd/api: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/tools/clientcmd/api/v1: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/tools/metrics: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/tools/reference: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/transport: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/util/cert: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/util/connrotation: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/util/flowcontrol: in Gopkg.lock's input-imports, but neither imported nor required
k8s.io/client-go/util/integer: in Gopkg.lock's input-imports, but neither imported nor required
sigs.k8s.io/yaml: in Gopkg.lock's input-imports, but neither imported nor required
```

@humblec can you please check

**Comment by humblec** [comment]
@Madhu-1 checking.

**Comment by j-griffith** [comment]
I'm not sure where dep is getting into a cycle here, but I was able to fix this up by:
1. Check out this PR
2. Run `dep ensure -update github.com/container-storage-interface/spec`

That cleaned things up and dep check looks good now.  There are a number of now unused packages in vendor it seems, but any clean up on those should probably be a separate effort, or just consider not committing vendor at all (I realize there are pros and cons either way on that one)

**Comment by humblec** [comment]
@j-griffith true, there is some cycle going on with dependency chain. The dep check does not fail now, but i am wondering about the vendor part. Even I am not interested to checkin `vendor` in the repo, but there are pros and cons.

**Comment by humblec** [comment]
The mystery is that, eventhough we can see `dep` wrote updates to vendor, `git diff` doesnt show any changes in the repo.  I tried 4/5 times since last week without luck.

**Comment by Madhu-1** [comment]
@humblec I have updated my old PR https://github.com/ceph/ceph-csi/pull/299   can you please review it, it's needed for me to update e2e code.

**Comment by humblec** [comment]
>@humblec I have updated my old PR #299 can you please review it, it's needed for me to update e2e code.
>
Sure, however I still dont know why `git diff` is not showing the vendor update eventhough the console list them. First I thought its specific to my system, but its not the case as its happening in other setup too like yours.. Its still a mystery for me. But, without delay lets get OLD PR in. I will experiment more on this later.  Thanks!

**Comment by humblec** [comment]
Closing this in favor of #299 299

**Pull request closed without merging**